pkgs/by-name/om/omada-sdn-controller/package.nix

```diff
@@ -0,0 +1,149 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  makeWrapper,
+  writeScriptBin,
+  coreutils,
+  curl,
+  jsvc,
+  openjdk17,
+  mongodb-ce,
+  omada-sdn-controller,
+  testers,
+  writeShellScript,
+  common-updater-scripts,
+  pup,
+  gnugrep,
+  gnused,
+  dataDir ? "/var/lib/omada",
+}:
+
+let
+  # The 'arch' command is required during runtime with is equivalent to 'uname -m':
+  arch = writeScriptBin "arch" "uname -m";
+
+  # Java 17 is supported, pin to avoid unsupported dependency upgrades.
+  jre = openjdk17;
+
+  # MongoDB 7 is supported with version 5.14.20 and above, pin to avoid unsupported dependency upgrades.
+  mongodb = mongodb-ce.overrideAttrs (oldAttrs: {
+    version = "7.0.14";
```

Comment:
That does nothing other than overwriting metadata.

Reply:
Okay... so how would it be correct? 😄

Reply:
you also need to overwrite src